contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,451 @@
+        _setSurplusTarget(_surplusTarget);
+        _setContractAdminRole(keccak256("PSM_ADMIN_ROLE"));
+    }
+
+    /// @notice modifier that allows execution when redemptions are not paused
+    modifier whileRedemptionsNotPaused() {
+        require(!redeemPaused, "PegStabilityModule: Redeem paused");
+        _;
+    }
+
+    /// @notice modifier that allows execution when minting is not paused
+    modifier whileMintingNotPaused() {
+        require(!mintPaused, "PegStabilityModule: Minting paused");
+        _;
+    }
+
+    /// @notice set secondary pausable methods to paused
+    function pauseRedeem() external onlyGovernorOrGuardianOrAdmin {
+        redeemPaused = true;
+        emit RedemptionsPaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to unpaused
+    function unpauseRedeem() external onlyGovernorOrGuardianOrAdmin {
+        redeemPaused = false;
+        emit RedemptionsUnpaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to paused
+    function pauseMint() external onlyGovernorOrGuardianOrAdmin {
+        mintPaused = true;
+        emit MintingPaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to unpaused
+    function unpauseMint() external onlyGovernorOrGuardianOrAdmin {
+        mintPaused = false;
+        emit MintingUnpaused(msg.sender);
+    }
+
+    /// @notice withdraw assets from PSM to an external address
+    function withdraw(address to, uint256 amount)
+        external
+        virtual
+        override
+        onlyPCVController
+    {
+        _withdrawERC20(address(underlyingToken), to, amount);
+    }
+
+    /// @notice set the ideal amount of reserves for the contract to hold for redemptions
+    function setReservesThreshold(uint256 newReservesThreshold)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setReservesThreshold(newReservesThreshold);
+    }
+
+    /// @notice set the target for sending surplus reserves
+    function setSurplusTarget(IPCVDeposit newTarget)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setSurplusTarget(newTarget);
+    }
+
+    /// @notice helper function to set reserves threshold
+    function _setReservesThreshold(uint256 newReservesThreshold) internal {
+        require(
+            newReservesThreshold > 0,
+            "PegStabilityModule: Invalid new reserves threshold"
+        );
+        uint256 oldReservesThreshold = reservesThreshold;
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdUpdate(
+            oldReservesThreshold,
+            newReservesThreshold
+        );
+    }
+
+    /// @notice helper function to set the surplus target
+    function _setSurplusTarget(IPCVDeposit newSurplusTarget) internal {
+        require(
+            address(newSurplusTarget) != address(0),
+            "PegStabilityModule: Invalid new surplus target"
+        );
```

Comment:
This should include a check that the NCPSM includes which checks that the underlying token of the new pcv deposit is the same as the underlying token of this PSM.